core/src/main/java/io/micronaut/core/bind/ArgumentBinderRegistry.java

```diff
@@ -31,20 +31,19 @@
      * Adds a request argument binder to the registry.
      * @param binder The binder
      * @param <T> The argument type
-     * @param <ST> The source type
-     * @since 2.0
+     * @since 4.0.0
      */
-    default <T, ST> void addRequestArgumentBinder(ArgumentBinder<T, ST> binder) {
+    default <T> void addArgumentBinder(ArgumentBinder<T, S> binder) {
         throw new UnsupportedOperationException("Binder registry is not mutable");
     }
 
     /**
-     * Locate an {@link ArgumentBinder} for the given argument and source type.
+     * Locate an {@link ArgumentBinder} for the given argument.
      *
      * @param argument The argument
-     * @param source   The source
      * @param <T>      The argument type
      * @return An {@link Optional} of {@link ArgumentBinder}
+     * @since 4.0.0
      */
-    <T> Optional<ArgumentBinder<T, S>> findArgumentBinder(Argument<T> argument, S source);
+    <T> Optional<ArgumentBinder<T, S>> findArgumentBinder(Argument<T> argument);
```

Comment:
keep this method and add a new default one  that passes `null` to minimize the breakage

Reply:
Reverted and made them deprecated